test/NetTopologySuite.Samples.Console/Technique/DatelineUtility.cs

```diff
@@ -0,0 +1,233 @@
+            // to list in this routine so the more popular SRIDs will be added as needed.
+            System.Diagnostics.Debug.Print("srid={0}", geometry.SRID);
+            return geometry.SRID == 4326;
+        }
+
+        /// <summary>
+        /// This is a filter that attempts to determine if the geometry crosses the dateline. Essentially, this is the key to the
+        /// entire dateline splitting utility.
+        /// </summary>
+        private class IsCrossesDatelineFilter : IGeometryFilter
+        {
+            public bool Done => IsCrossesDateline;
+
+            public bool GeometryChanged => false;
+
+            public bool IsCrossesDateline { get; private set; }
+
+            public void Filter(Geometry geometry)
+            {
+                // since we are filtering per geometry we could have multiple parts. so we must test each part individually for traversing
+                // dateline. if we were to test the CoordinateSequence with a IEntireCoordinateSequenceFilter filter then we could get false
+                // positives (ex: two islands on each side of the dateline).
+                System.Diagnostics.Debug.Print("num geoms={0}", geometry.NumGeometries);
+                Geometry part = null;
+                for (int i = 0; i < geometry.NumGeometries; i++)
+                {
+                    part = geometry.GetGeometryN(i);
+                    System.Diagnostics.Debug.Print("  part {0} of type {1}", i, part.GeometryType);
+
+                    // for each part we cycle through segments in the coordinate sequence and determine if any cross the dateline. there will
+                    // be several scenarios that we will look for that determine if a segment traverses the dateline. the first is to check if
+                    // any longitudes < +180° and also > +180°. this will be an indication that the input coordinates are based on longitude
+                    // values in the range 0° to +360°. thus any longitudes that traverse the dateline will do so if they are on BOTH sides
+                    // of the +180° meridian.
+                    //double currX = seq.GetX(0);
+                    //bool wasLT = currX < -180d;
+                    //bool wasGT = currX > -180d;
+                    //for (int i = 1; i < seq.Count; i++)
+                    //{
+                    //    currX = seq.GetX(i);
+                    //    if (currX < -180d && wasGT ||
+                    //        currX > -180d && wasLT)
+                    //    {
+                    //        IsCrossingDateline = true;
+                    //        break;
+                    //    }
+                    //    if (currX < -180d) wasLT = true;
+                    //    else if (currX > -180d) wasGT = true;
+                    //}
+
+                    bool hasLonsLt180 = part.Coordinates.Any(c => c.X < 180);
```

Comment:
Don't work on `Geometry.Coordinates`, that is potentially a bottleneck. For this check you iterate over the whole array twice + you might need to build the array of coordinates twice. Use `IEntireCoordinateSequenceFilter` to do that check.